test/asynchronous/test_client.py

```diff
@@ -57,6 +60,7 @@
     unittest,
 )
 from test.asynchronous.pymongo_mocks import AsyncMockClient
+from test.test_binary import TestBinary
```

Comment:
Importing TestBinary directly will actually cause pytest/unittest to pick up and run TestBinary in this file too. Instead we need to mangle to name, like: `from test.test_binary import TestBinary as BinaryBase` or move the test data out of TestBinary into to a global var that we can import.